hare/mock_oracle.go

```diff
@@ -1,8 +1,9 @@
 package hare
```

Comment:
Why is this file called "mock_oracle.go". There is a lot of non-mock code in it...

Reply:
Because MockHashOracle is a mock.
Do you want me to move expected func and Rolacle interface to another file? ain't critic IMO

Reply:
Please do, it is better to differentiate between mock code and production code

Reply:
Done